domains/certificates/Query.API/API/DataSyncSyncer/DataSyncSyncerWorker.cs

```diff
@@ -39,6 +45,20 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
             var allContracts = await GetAllContracts(stoppingToken);
             foreach (var contract in allContracts)
             {
+                if (string.IsNullOrEmpty(contract.PublicKey))
```

Comment:
Have you considered going DDD on CertificateIssuingContract (contract) and other domain objects?